### DISCUSSION TITLE
Redirecting from mount in a nested component does not work and throw an error.

### DISCUSSION THREAD

**Discussion by QuentinGab** [livewire/livewire#6815] [7. Bugs (archived)]
### Livewire version

v3.0.3

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

 1. Create a livewire component ParentComponent that render a child Livewire component ChildComponent
 2. Redirect to anything in the mount method of ChildComponent using `$this->redirect(...)` or `redirect()`
 3. It throws `Object of class Livewire\Features\SupportRedirects\Redirector could not be converted to int`.
 

### Additional information

Redirecting from the mount method of ParentComponent works as intended.

If it's a limitation, it may be useful to mention it in the doc at the page [Redirecting](https://livewire.laravel.com/docs/redirecting#basic-usage).

I would like to be able to do such redirection in the mount method in case a resource is not found for example.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by alpaltunel** [comment]
Yes same here. even though I put a redirect to constructor it fails.
I put the @livewire('top-menu') to dashboard-layout.blade.php and that top-menu requires session object to be set.
so I got an error: Object of class Livewire\Features\SupportRedirects\Redirector could not be converted to int

**Comment by titonova** [comment]
Same issue here. I ended up just using the manual PHP header('Location: '.route('...')) method

**Comment by lrljoe** [comment]
From a usability perspective, that'd be pretty annoying, and has limited use cases imho.

If you have a full page component, with lets say 4 child components, one of which has some broken data, you're suggesting to throw the user somewhere else, instead of handling the issue with that particular child component gracefully, and allowing them to work with the other components on the page.

**Reply by titonova** [reply]
What if you aren't trying to redirect because of broken data though? What if its intentional, that after an action is performed in a child component, the user should get redirected to another page to say view/complete it etc

**Comment by Mohammadtrabelsi** [comment]
Same issue here.

**Reply by lrljoe** [reply]
@Mohammadtrabelsi 
Rather than reply to a 2 year old post, please create a new one along with relevant information as to error messages etc, and what versions are in play.
If you're still using 3.0.3, which this thread discusses, then it's no wonder you're having issues.
